apps/lib/app_provider.c

```diff
@@ -41,6 +44,43 @@ static int opt_provider_load(const char *provider)
     return 1;
 }
 
+/*
+ * Add this call to the start of any application that support a library context.
+ * NOTE: app_load_default_provider() should also be called after the options
+ * have been processed.
+ */
+OPENSSL_CTX *app_new_libctx(void)
+{
+    /*
+     * Load the NULL provider into the default library context and create a
+     * library context which will then be used for any OPT_PROV options.
+     */
+    if (libctx == NULL) {
+        if (!opt_provider_load("null"))
+            return NULL;
+        libctx = OPENSSL_CTX_new();
+    }
+    return libctx;
+}
+
+int app_load_default_provider(void)
+{
+    int count = 0;
+    void *data = &count;
+
+    if (libctx == NULL)
+        return 1;
+
+    if (!OSSL_PROVIDER_do_all(libctx, provider_count, data))
+        return 0;
+
+    if (count == 0) {
+        if (!opt_provider_load("default"))
+            return 0;
+    }
```

Comment:
Why do you do this, for something that should work through fallbacks?

Reply:
Err assuming that the NULL context has been loaded with the NULL provider I thought this will not work.

Reply:
This code won't work either, then, will it?  The "null" provider counts as *one*, doesn't it?  So what I see you doing here is to load the "default" provider when no other providers (not even the "null" provider) have been loaded, which is exactly what the fallback mechanism does, no?

Reply:
Oh, hold on, `opt_provider_load` only touches the default library context.  Quite frankly, though, I have a hard time understanding what you want to accomplish with this...  after all, it's not like doing anything with a non-default library context will magically fall back on whatever's in the default provider, will it?  At least conceptually...

Reply:
Removed this code - you are right the fallback would handle it.